ui/components/app/alerts-system/utils.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  BlockaidResultType,
+  SecurityProvider,
+} from '../../../../shared/constants/security-provider';
+import {
+  Alert,
+  AlertSeverity,
+} from '../../../ducks/confirm-alerts/confirm-alerts';
+import {
+  BackgroundColor,
+  Severity,
+} from '../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import {
+  REASON_TO_DESCRIPTION_TKEY,
+  REASON_TO_TITLE_TKEY,
+} from '../../../pages/confirmations/components/security-provider-banner-alert/blockaid-banner-alert/blockaid-banner-alert';
+import { SecurityAlertResponse } from '../../../pages/confirmations/types/confirm';
+import { BannerAlertSeverity } from '../../component-library';
+
+/**
+ * Returns the background color based on the severity level.
+ *
+ * @param severity - The severity level.
+ * @returns The background color corresponding to the severity level.
+ */
+export function getSeverityBackground(severity: Severity): BackgroundColor {
+  switch (severity) {
+    case Severity.Danger:
+      return BackgroundColor.errorMuted;
+    case Severity.Warning:
+      return BackgroundColor.warningMuted;
+    default:
+      return BackgroundColor.primaryMuted;
+  }
+}
+
+/**
+ * Returns the highest  severity from an array of alerts.
+ *
+ * @param alerts - An array of alerts.
+ * @returns The highest  severity.
+ */
+export function getHighestSeverity(alerts: Alert[]): AlertSeverity {
+  return alerts.reduce((highestSeverity: AlertSeverity, alert: Alert) => {
+    if (alert.severity === Severity.Danger) {
+      return Severity.Danger;
+    } else if (
+      alert.severity === Severity.Warning &&
+      highestSeverity !== Severity.Danger
+    ) {
+      return Severity.Warning;
+    }
+    return highestSeverity;
+  }, Severity.Info);
+}
+
+/**
+ * Converts the severity of a banner alert to the corresponding BannerAlertSeverity.
+ *
+ * @param severity - The severity of the banner alert.
+ * @returns The corresponding BannerAlertSeverity.
+ */
+export function getBannerAlertSeverity(
+  severity: Severity,
+): BannerAlertSeverity {
+  switch (severity) {
+    case Severity.Danger:
+      return BannerAlertSeverity.Danger;
+    case Severity.Warning:
+      return BannerAlertSeverity.Warning;
+    default:
+      return BannerAlertSeverity.Info;
+  }
+}
+
+/**
+ * Returns the corresponding AlertSeverity based on the provided BlockaidResultType.
+ *
+ * @param severity - The BlockaidResultType to determine the AlertSeverity for.
+ * @returns The AlertSeverity corresponding to the provided BlockaidResultType.
+ */
+export function getProviderAlertSeverity(
```

Comment:
Since we're aiming for this alert system to be decoupled from confirmations and any usage, then should this method not exist outside of it since it's coupled to Blockaid which is a specific usage of the alerts?

Perhaps inside `usePersonalSignAlerts` or a related util?